packages/wallets/avana/src/adapter.ts

```diff
@@ -143,7 +148,10 @@ export class AvanaWalletAdapter extends BaseMessageSignerWalletAdapter {
     async disconnect(): Promise<void> {
         const wallet = this._wallet;
         if (wallet) {
-            wallet.off('disconnect', this._disconnected);
+            if (this._onDisconnectBound) {
+                wallet.off('disconnect', this._onDisconnectBound);
+                this._onDisconnectBound = null;
+            }
```

Comment:
This merely complicates the existing check since we are relying on the checks in the `connect` function to guarantee only having one listener. The key thing is to prevent multiple listeners from being added. Is there a reason to believe we aren't doing that in all cases?